indexing-service/src/main/java/org/apache/druid/indexing/input/DruidInputSource.java

```diff
@@ -87,13 +91,21 @@
   @Nullable
   private final List<WindowedSegmentId> segmentIds;
   private final DimFilter dimFilter;
```

Comment:
will it make sense to move DimFilter outside the InputSource in the task json? It seems more natural to me to put the filters alongside transforms, dimensions, and metrics and leave only the data source properties inside the `InputSource` section. On the flip side, it could make the compatibility situation more complicated than it is.

Reply:
It's possible to specify a filter alongside transforms today! You can do it in two places:

- In the `transformSpec` (this works with any input source / format, see https://druid.apache.org/docs/latest/ingestion/index.html#filter)
- In the druid `inputSource` itself (of course, only works with this input source)

It's a little silly to have both, perhaps, but there's a practical reason: specifying a filter in the druid `inputSource` is faster, because it is applied while creating the cursor that reads the data, and therefore it can use indexes, etc. The filter in the `transformSpec` is applied _after_ the cursor generates rows.

But I think in the future, it'd be better to support pushing down the `transformSpec` filter into the cursor, and then we could deprecate the filter parameter in the `inputSource`, because it wouldn't be useful anymore.

For now, I suggest we leave it as-is.